### PR TITLE
Improve synonym functionality

### PR DESCRIPTION
**Pull request by loevgaard**
## Index

image

## Create

image

## Update

image

### CODE REVIEW

File: src/Repository/SynonymRepository.php [hunk 1]
```diff
@@ -18,8 +18,7 @@ public function findByLocaleAndChannel(string $localeCode, string $channelCode =
         ;
 
         if (null !== $channelCode) {
-            $qb->leftJoin('o.channel', 'c')
-                ->andWhere('o.channel IS NULL OR c.code = :channelCode')
+            $qb->join('o.channels', 'c', 'WITH', 'c.code = :channelCode')
```
Comment (loevgaard) [inline src/Repository/SynonymRepository.php:21]: The repository method also needs to take care of the chosen indexes

File: src/EventListener/Doctrine/SynonymListener.php [hunk 2]
```diff
@@ -30,12 +34,31 @@ public function postRemove(LifecycleEventArgs $eventArgs): void
...
+            return;
+        }
+
         $this->commandBus->dispatch(new UpdateSynonyms());
     }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::RESPONSE => ['dispatch', 10],
```
Comment (loevgaard) [inline src/EventListener/Doctrine/SynonymListener.php:61]: @Zales0123 @jakubtobiasz: You you also listen to the `RESPONSE` or would you go for another event?

Comment (jakubtobiasz) [reply src/EventListener/Doctrine/SynonymListener.php:61]: Why don't you listen to resource events?

Comment (loevgaard) [reply src/EventListener/Doctrine/SynonymListener.php:61]: I am not sure I know what resource events to listen to? I mean the indexes are configurable, so I don't know the entities/resources used to build an index.

Comment (loevgaard) [reply src/EventListener/Doctrine/SynonymListener.php:61]: Are they also fired when using the API?

**Review by jakubtobiasz — APPROVED** [review]

**Pull request merged**